numpy/core/arrayprint.py

```diff
@@ -349,28 +338,26 @@ def _get_format_function(data, precision, suppress_small, formatter):
     else:
         return formatdict['numpystr']()
 
-def _array2string(a, max_line_width, precision, suppress_small, separator=' ',
-                  prefix="", formatter=None):
+def _array2string(a, options, separator=' ', prefix=""):
 
-    if a.size > _summaryThreshold:
+    if a.size > options['threshold']:
         summary_insert = "..., "
         data = _leading_trailing(a)
     else:
         summary_insert = ""
-        data = ravel(asarray(a))
+        data = ravel(asarray(a)) if a.shape != () else asarray(a)
```

Comment:
This is kind of weird, and probably deserves a comment. Perhaps the special case should be size==1, not ndim==0, which eliminates the need for this line